### DISCUSSION TITLE
Retrieving db_ports as timing path starting points in OpenROAD

### DISCUSSION THREAD

**Discussion by liangrj2014** [The-OpenROAD-Project/OpenROAD#5499] [General]
Timing path starting points can be either the CLK pins of registers or input ports. By using the following commands, we can retrieve all the OpenROAD db_pins that serve as starting points:

set startpoints [::sta::startpoints]
foreach startpoint $startpoints {
  set start_point_pin [::sta::sta_to_db_pin $startpoint]
  if { $start_point_pin == "NULL" } {
    continue
  }
  ....
}

However, I have encountered an issue where we are unable to obtain input ports using similar commands. The sta_to_db_port command referenced [here] (https://github.com/The-OpenROAD-Project/OpenROAD/blob/b565cb9323452d559c35016b6412437d1609b1ca/src/dbSta/src/dbSta.i#L99) does not seem to function as we expected.

Could you kindly suggest an alternative method to retrieve db_ports that are starting points? Thanks in advance!

**Comment by maliberty** [comment]
Could you give a small test case showing the issue?

**Comment by maliberty** [comment]
Solved with https://github.com/The-OpenROAD-Project/OpenROAD/pull/5559

**Reply by liangrj2014** [reply]
Great! Thanks!
